### PR TITLE
chore(deps): update to stencil v3

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Ionic Framework is using Stencil 2

## What is the new behavior?

- Updates core to using Stencil v3
- Disabled sourceMaps until the related stencil bug can be resolved. (We are not generating sourcemaps in Ionic v6, so there should be no behavior change)
- Updated sync scripts to account for dependencies (see https://github.com/ionic-team/ionic-framework/pull/26686)
- Refactor so internal-only interfaces are imported from the files directly instead of the `interfaces.d.ts` file
- Updates `interfaces.d.ts` file so only public interfaces are explicitly exported
- Updated the output target dependencies to account for the Stencil v3 dep

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

Note: I discovered that we had a dozen or so private interfaces exported as a result of how the `interfaces.d.ts` file is structured in `main`. This PR makes it so those interfaces are no longer exported. However, I am not considering this a breaking change since these are private APIs and should not be used in external apps.

### CODE REVIEW

File: core/src/components/datetime/datetime-interface.ts [hunk 1]
```diff
@@ -1,7 +1,3 @@
-export interface DatetimeOptions {
```
Comment (liamdebeasi) [inline core/src/components/datetime/datetime-interface.ts:1]: This is not used anywhere

File: core/src/components/spinner/spinner-interface.ts [hunk 2]
```diff
@@ -1,5 +1,3 @@
-export { SpinnerTypes } from './spinner-configs';
```
Comment (liamdebeasi) [inline core/src/components/spinner/spinner-interface.ts:1]: SpinnerTypes is already exported by Stencil

File: core/stencil.config.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ import { apiSpecGenerator } from './scripts/api-spec-generator';
 
 export const config: Config = {
   autoprefixCss: true,
+  sourceMap: false,
```
Comment (liamdebeasi) [inline core/stencil.config.ts:12]: There is a bug in stencil that causes sourcemap generation to fail in Ionic. The stencil team is aware of this and they are going to fix it.

File: core/src/components/button/button.tsx [hunk 4]
```diff
@@ -2,12 +2,15 @@ import type { ComponentInterface, EventEmitter } from '@stencil/core';
...
-import type { AnimationBuilder, Color, RouterDirection } from '../../interface';
+import type { AnimationBuilder, Color } from '../../interface';
 import type { AnchorInterface, ButtonInterface } from '../../utils/element-interface';
 import type { Attributes } from '../../utils/helpers';
 import { inheritAriaAttributes, hasShadowDom } from '../../utils/helpers';
 import { printIonWarning } from '../../utils/logging';
 import { createColorClasses, hostContext, openURL } from '../../utils/theme';
+import type { RouterDirection } from '../router/utils/interface';
+
+0;
```
Comment (amandaejohnston) [inline core/src/components/button/button.tsx:13]: I'm guessing this `0` slipped in by mistake? 😆

**Review by amandaejohnston — APPROVED** [review]
Great work! Confirmed that I'm still able to build and run core on my Windows machine too 👍

File: core/src/components/button/button.tsx (hunk 4 above)
Comment (liamdebeasi) [reply core/src/components/button/button.tsx:13]: 👀 yes

File: angular/src/directives/angular-component-lib/utils.ts [hunk 5]
```diff
@@ -4,46 +4,40 @@ import { fromEvent } from 'rxjs';
 
 export const proxyInputs = (Cmp: any, inputs: string[]) => {
   const Prototype = Cmp.prototype;
-  inputs.forEach(item => {
+  inputs.forEach((item) => {
```
Comment (rwaskiewicz) [inline angular/src/directives/angular-component-lib/utils.ts:7]: Double checking....should these format changes be in this PR? I didn't see a callout about it in the summary || any prettier changes in `package.json`

Comment (liamdebeasi) [reply angular/src/directives/angular-component-lib/utils.ts:7]: Good catch -- I added this to the original PR. I updated the output target deps, and it looks like there were some formatting changes: https://github.com/ionic-team/stencil-ds-output-targets/blob/f20a7780bf6416052e338fd31bd1a813477f9792/packages/angular-output-target/angular-component-lib/utils.ts#L7

**Review by rwaskiewicz — APPROVED** [review]

**Review by sean-perkins — APPROVED** [review]
These changes look good to me. I had the same question as Ryan, regarding the `@utils` alias import paths for the animations file.

**Comment by liamdebeasi** [comment]
Good catch. I reverted those changes in https://github.com/ionic-team/ionic-framework/pull/26663/commits/45c4ff49c9c85b9d85ae74a99fd07cbe0d7d75ac. Also sorry to the person I unintentionally tagged in my commit message 😅.

File: core/src/utils/menu-controller/animations/base.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
-import type { Animation } from '../../../interface';
-import { createAnimation } from '../../animation/animation';
+import { createAnimation } from '@utils/animation/animation';
```
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply core/src/utils/menu-controller/animations/base.ts]: Good catch - reverted that https://github.com/ionic-team/ionic-framework/pull/26663/commits/45c4ff49c9c85b9d85ae74a99fd07cbe0d7d75ac

**Pull request merged**